### PR TITLE
docs: Update eks managed node group usage to v19

### PR DESCRIPTION
**Pull request by don-attilio**
## Description
Minimal update in documentation in order to align it to version 19.x.x.
I added a sentence on the implications of using custom launch template, which could have saved me quite some time, but feel free to close this PR if not necessary.

## Motivation and Context
Just more ease of use I guess.

## Breaking Changes
Changes do not affect existing code.

## How Has This Been Tested?
Changes do not affect existing code.

### CODE REVIEW

File: docs/compute_resources.md [hunk 1]
```diff
@@ -28,8 +28,6 @@ Refer to the [EKS Managed Node Group documentation](https://docs.aws.amazon.com/
 ```hcl
   eks_managed_node_groups = {
     bottlerocket_default = {
-      use_custom_launch_template = false
```
Comment (don-attilio) [inline docs/compute_resources.md:31]: Line deleted as I feel it is a bit misleading and unrelated to how to use Bottlerocket.

Comment (bryantbiggs) [reply docs/compute_resources.md:31]: this is intended - this is showing using the bottlerocket AMI with the default EKS managed node group launch template

Comment (bryantbiggs) [reply docs/compute_resources.md:31]: revert this change and we can merge the doc changes proposed

Comment (don-attilio) [reply docs/compute_resources.md]: Done!

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 19.4.3 :tada:

**Pull request merged**